ls1intum/static-code-analysis-parser
# Feature/add c language support

### Checklist
- [x] I added a integration test for the newly added tool.
- [x] I commented the code.

### Motivation and Context


Adds support for GCC as a static analysis tool. A feature the operating systems lecture would like to use next semester in order to make the feedback more readable and compact.

### Description

I added a Parser for the GCC tools, that makes use of regex. Unfortunately, GCC does not offer XML export of reports. The output of GCC  gets wrapped in `root` tags, then gets splitted with respect to beginnings of new warnings. The warnings then get categorized by hand to be more useful to the student, while enabling instructors to disable certain categories in case of issues. Furthermore, I added an example file with a corresponding test.

I've also seen that Spotless enforces a space between each field for my code. I've rarely seen this. Is this correct?

### Steps for Testing


1. Replace the the _static-code-analysis-parser_ `.jar` with the one obtained by compiling and packing this branch in the `.m2/repository` folder.
2. Build the Bamboo Server Plugin with `atlas-package` and upload it to Bamboo using `Manage apps`.
3. Run the [corresponding Artemis branch](https://github.com/ls1intum/Artemis/tree/feature/add-c-sca-support) locally
4. Navigate to Artemis and create a C programming exercise with SCA enabled. 
5. Create some issue like a double free error in the code, commit and push it, then check out the result in Artemis.
